src/main/java/org/jabref/logic/formatter/bibtexfields/EscapeCurrencySymbolsFormatter.java

```diff
@@ -0,0 +1,37 @@
+package org.jabref.logic.formatter.bibtexfields;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+
+import org.jabref.logic.cleanup.Formatter;
+import org.jabref.logic.l10n.Localization;
+
+public class EscapeCurrencySymbolsFormatter extends Formatter {
+
+    @Override
+    public String getName() {
+        return Localization.lang("Escape currency symbols");
+    }
+
+    @Override
+    public String getKey() {
+        return "escapeCurrencySymbols";
+    }
+
+    @Override
+    public String format(String value) {
+        Objects.requireNonNull(value);
+        String replacement = "\\$";
+        return value.replaceAll("(?<!\\\\)\\$", Matcher.quoteReplacement(replacement));
```

Comment:
I think, here a double replacement is introduced

```suggestion
        return value.replaceAll("(?<!\\\\)\\$", "\\$");
```

Reply:
Seems that `value.replaceAll("(?<!\\\\)\\$", "\\$") ` won't change $ to \\$ , $ will not be changed
`value.replaceAll("(?<!\\\\)\\$", "$")` and `value.replaceAll("(?<!\\\\)\\$", "\\\\$")`  will result in `java.lang.IllegalArgumentException: Illegal group reference: group index is missing`,
so I believe that it's mecessary to use:
`value.replaceAll("(?<!\\\\)\\$", Matcher.quoteReplacement(replacement));`